### ISSUE TITLE
[PALADIN] Hand of Sacrifice vs Spellsteal

### ISSUE THREAD

**Issue by haephaistoss** [gamefreedomgit/Frostmourne#1650]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (https://www.wowhead.com/wotlk)

**Description:** Currently if a mage spellsteals Sac, no dmg is split anymore, no dmg is transferred to the paladin.

**How to reproduce:** Log in with 3 characters, sac a friendly target, have your mage spellsteal it, and hit the mage with the spellstolen buff.

**How it should work:** Spellstealing Sac should keep splitting the dmg to the caster paladin, the only difference being, that after being stolen the base of the splitting is not the target the original sac was cast on, but the mage, who has stolen it.

So to sum it up:

The buff only determines the source of the split, the target of the split dmg is ALWAYS the paladin who originally cast the spell.

That's why mages usually macro it into spellsteal to cancel sac aura, otherwise the moment they take any dmg, paladin is out of cc, so spellstealing it is pointless without canceling.

**Database links:** https://www.wowhead.com/wotlk/spell=6940/hand-of-sacrifice

https://cata-twinhead.twinstar.cz/?issue=11204
https://wow.gamepedia.com/Hand_of_Sacrifice
http://www.skill-capped.com/forums/showthread.php?30966-Can-someone-explain-a-Mage-s-Spellsteal&p=17

**Comment by haephaistoss** [comment]
Testing on the PTR, it doesn't work anymore.

Author: Piootrek 
Message: Core/Spells: Stealer should be caster of stolen spell
https://github.com/gamefreedomgit/Frostmourne/issues/1362 

This fix prolly broke it

**Comment by haephaistoss** [comment]
https://www.warcraftmovies.com/movieview.php?id=55762

2:40

**Comment by Piootrek** [comment]
Fixed
